src/main/java/org/jenkinsci/plugins/github_branch_source/IssueCommentGHEventSubscriber.java

```diff
@@ -0,0 +1,220 @@
+    protected Set<GHEvent> events() {
+        return immutableEnumSet(ISSUE_COMMENT);
+    }
+
+    /**
+     * Handles comments on pull requests.
+     * @param event only ISSUE_COMMENT event
+     * @param payload payload of gh-event. Never blank
+     */
+    @Override
+    protected void onEvent(GHEvent event, String payload) {
+        JSONObject json = JSONObject.fromObject(payload);
+
+        // Make sure this issue is a PR
+        final String issueUrl = json.getJSONObject("issue").getString("html_url");
+        Matcher matcher = PULL_REQUEST_ID_PATTERN.matcher(issueUrl);
+        if (!matcher.matches()) {
+            LOGGER.log(Level.FINE, "Issue comment is not for a pull request, ignoring {0}", issueUrl);
+            return;
+        }
+
+        final String pullRequestId = matcher.group(1);
+        final String pullRequestJobName = "PR-" + pullRequestId;
+        
+        // Verify that the comment body matches the trigger build string
+        final String commentBody = json.getJSONObject("comment").getString("body");
+        final String commentUrl = json.getJSONObject("comment").getString("html_url");
+
+        // Make sure the action is edited or created (not deleted)
+        String action = json.getString("action");
+        if (!ACTION_CREATED.equals(action) && !ACTION_EDITED.equals(action)) {
+            LOGGER.log(Level.FINER, "Issue comment action is not created or edited ({0}) for PR {1}",
+                new Object[] { action, issueUrl }
+            );
+            return;
+        }
+
+        // Make sure the repository URL is valid
+        String repoUrl = json.getJSONObject("repository").getString("html_url");
+        matcher = REPOSITORY_NAME_PATTERN.matcher(repoUrl);
+        if (!matcher.matches()) {
+            LOGGER.log(Level.WARNING, "Malformed repository URL {0}", repoUrl);
+            return;
+        }
+        final GitHubRepositoryName changedRepository = GitHubRepositoryName.create(repoUrl);
+        if (changedRepository == null) {
+            LOGGER.log(Level.WARNING, "Malformed repository URL {0}", repoUrl);
+            return;
+        }
+
+        LOGGER.log(Level.FINE, "Received comment on PR {0} for {1}", new Object[] { pullRequestId, repoUrl });
+        ACL.impersonate(ACL.SYSTEM, new Runnable() {
+            @Override
+            public void run() {
+                boolean jobFound = false;
+                for (final SCMSourceOwner owner : SCMSourceOwners.all()) {
+                    for (SCMSource source : owner.getSCMSources()) {
+                        if (!(source instanceof GitHubSCMSource)) {
+                            continue;
+                        }
+                        GitHubSCMSource gitHubSCMSource = (GitHubSCMSource) source;
+                        if (gitHubSCMSource.getRepoOwner().equals(changedRepository.getUserName()) &&
+                                gitHubSCMSource.getRepository().equals(changedRepository.getRepositoryName())) {
```

Comment:
Be careful with case-sensitive comparisons: [JENKINS-39114](https://issues.jenkins-ci.org/browse/JENKINS-39114)